packages/kbn-shared-gen-ai-ui/components/providers/get_providers.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { HttpSetup } from '@kbn/core-http-browser';
+import { i18n } from '@kbn/i18n';
+import { useQuery } from '@tanstack/react-query';
+import type { ToastsStart } from '@kbn/core-notifications-browser';
+import { InferenceProvider } from '../../types';
+import { INTERNAL_BASE_GEN_AI_API_PATH } from '../../constants';
+
+export const getProviders = async (http: HttpSetup): Promise<InferenceProvider[]> => {
+  return await http.get(`${INTERNAL_BASE_GEN_AI_API_PATH}/_inference/_services`);
```

Comment:
Do we want to share the same API endpoints between plugins that use this package?

Abstracting the hooks with API calls will create a direct dependency on the `stack_connectors` plugin for any plugin that needs to use this package.

Did we consider abstracting just the components first?

The `search_inference_endpoints` plugin that will be using this package is used in both stack and serverless search.

Reply:
Discussed with @TattdCodeMonkey, and for now, the plan is to have the UI components focus solely on rendering, without handling any logic or making API calls. Initially, I assumed that packages could handle API calls, but it seems the best approach would be to create `lib` functions that can be shared between plugins instead.
cc: @YulNaumenko